diag_manager/fms_diag_axis_object.F90

```diff
@@ -0,0 +1,287 @@
+    INTEGER,            INTENT(in), OPTIONAL :: direction       !< Indicates the direction of the axis
+    INTEGER,            INTENT(in), OPTIONAL :: edges           !< Axis ID for the previously defined "edges axis"
+    TYPE(domain1d),     INTENT(in), OPTIONAL :: Domain          !< 1D domain
+    TYPE(domain2d),     INTENT(in), OPTIONAL :: Domain2         !< 2D domain
+    TYPE(domainUG),     INTENT(in), OPTIONAL :: DomainU         !< Unstructured domain
+    CHARACTER(len=*),   INTENT(in), OPTIONAL :: aux             !< Auxiliary name, can only be <TT>geolon_t</TT>
+                                                                !! or <TT>geolat_t</TT>
+    CHARACTER(len=*),   INTENT(in), OPTIONAL :: req             !< Required field names.
+    INTEGER,            INTENT(in), OPTIONAL :: tile_count      !< Number of tiles
+    INTEGER,            INTENT(in), OPTIONAL :: domain_position !< Domain position, "NORTH" or "EAST"
+
+    obj%axis_name = diag_copy_string(name)
+    obj%units = diag_copy_string(units)
+    obj%cart_name = diag_copy_string(cart_name) !< TO DO Check for valid cart_names
+    if (present(long_name)) obj%long_name = diag_copy_string(long_name)
+
+    select type (axis_data)
+    type is (real(kind=r8_kind))
+      allocate(real(kind=r8_kind) :: obj%axis_data(size(axis_data)))
+      obj%axis_data = axis_data
+    type is (real(kind=r4_kind))
+      allocate(real(kind=r4_kind) :: obj%axis_data(size(axis_data)))
+      obj%axis_data = axis_data
+    class default
+      call mpp_error(FATAL, "The axis_data in your diag_axis_init call is not a supported type. &
+                          &  Currently only r4 and r8 data is supported.")
+    end select
+
+    !< TO DO check the presence of multiple Domains
+    if (present(Domain)) then
+      allocate(fms_domain1d_t :: obj%axis_domain)
+      call obj%axis_domain%set(Domain=Domain)
+    else if (present(Domain2)) then
+      allocate(fms_domain2d_t :: obj%axis_domain)
+      call obj%axis_domain%set(Domain2=Domain2)
+    else if (present(DomainU)) then
+      allocate(fms_domainUG_t :: obj%axis_domain)
+      call obj%axis_domain%set(DomainU=DomainU)
+    endif
+
+    obj%tile_count = 1
+    if (present(tile_count)) obj%tile_count = tile_count
+
+    !< TO DO Check for valid domain_position
+    obj%domain_position = CENTER
+    if (present(domain_position)) obj%domain_position = domain_position
+
+    obj%length = size(axis_data)
+
+    !< TO DO Check for valid direction
+    obj%direction = 0
+    if (present(direction)) obj%direction = direction
+
+    !< TO DO Check if id is valid and with the same parameters
+    obj%edges = 0
+    if (present(edges)) obj%edges = edges
+
+    if (present(aux)) obj%aux = diag_copy_string(aux)
+    if (present(req)) obj%req = diag_copy_string(req)
+
+    obj%nsubaxis = 0
+  end subroutine diag_axis_init
+
+  !> @brief Get the length of the axis
+  !> @return axis length
```

Comment:
Imagine it like this:
```fortran
integer :: axis_length !< axis length
```
Would this be acceptable?  I mean... I guess it would be correct, but it seems a little too short.  Change it or don't, I don't care either way.  This is just nitpicky

Reply:
less is more

Reply:
smh